src/main/scala/org/broadinstitute/dsde/firecloud/utils/StreamingPassthrough.scala

```diff
@@ -0,0 +1,183 @@
+      extractRequest { req =>
+        complete {
+          routeResponse(req)
+        }
+      }
+    }
+  }
+
+
+  /**
+    * Accepts an http request from an end user to Orchestration,
+    * and converts it to a request suitable to send to a remote system
+    * for passthroughs:
+    *   - rewrite the URI to be valid for the remote system
+    *   - strip any http headers, such as Timeout-Access, that are invalid to send to the remote system
+    *
+    * This allows Orchestration to act as an API gateway,
+    * passing the user's request mostly untouched to another service.
+    *
+    * @param req the request inbound to Orchestration
+    * @return the outbound request to be sent to another service
+    */
+  def transformToPassthroughRequest(localBasePath: Uri.Path, remoteBaseUri: Uri)(req: HttpRequest): HttpRequest = {
+    // Convert the URI to the one suitable for the remote system
+    val targetUri = convertToRemoteUri(req.uri, localBasePath, remoteBaseUri)
+
+    // Remove unwanted headers. Akka automatically adds a Timeout-Access to the request
+    // for internal use in managing timeouts; see akka.http.server.request-timeout.
+    // This header is not legal to pass to an external remote system.
+    val targetHeaders = req.headers.filter(_.isNot(`Timeout-Access`.lowercaseName))
+
+    // TODO: what should this log?
+    streamingPassthroughLogger.info(s"${req.method} ${req.uri} => $targetUri")
+
+    // return a copy of the inbound request, using the new URI and new headers.
+    req.withUri(targetUri).withHeaders(targetHeaders)
+  }
+
+  /**
+    * Inspects the URI for an actual end-user request to Orchestration, finds the portion
+    * of that Uri after the localBasePath, and appends that remainder to the remoteBaseUri.
+    *
+    * The end result is a URI suitable to send to a remote system.
+    *
+    * @param requestUri the end-user's request to Orchestration
+    * @param localBasePath the portion of the path to strip off
+    * @param remoteBaseUri the remote system to use as passthrough target
+    * @return the URI suitable for sending to the remote system
+    */
+  def convertToRemoteUri(requestUri:Uri, localBasePath: Uri.Path, remoteBaseUri: Uri): Uri = {
+    // Ensure the incoming request starts with the localBasePath. Abort if it doesn't.
+    // This condition should only be caused by developer error in which the streamingPassthrough
+    // directive is incorrectly configured inside a route.
+    if (!requestUri.path.startsWith(localBasePath)) {
+      throw new Exception(s"request path doesn't start properly: ${requestUri.path} does not start with $localBasePath")
+    }
+
+    // recursively loops through the segments of the actual path, and removes
+    // all segments present in the base path. Returns the result:
+    // every part of the path, minus the base.
+    // This should be equivalent to `actual.toString.replace(base.toString, "")`
```

Comment:
I'm strongly in favor of the string conversion approach for readability. I'm not sure the recursive approach would be faster, since it's adding a bunch of function calls. My one concern with string conversion is that the solution will fail if the base path is for any reason duplicated in the middle of the path

Reply:
switched to string conversion, made sure to only replace the first instance, and added a unit test for this.